File: flask/models/rol_model.py
"""
Modelo para la tabla RolesOrganizador
"""
from config.database import Database

class RolModel:
    """Maneja operaciones de base de datos para Roles"""
    
    @staticmethod
    def get_all():
        """Obtiene todos los roles"""
        sql = "SELECT * FROM RolesOrganizador ORDER BY nombre"
        return Database.execute_query(sql, fetch=True) or []
    
    @staticmethod
    def get_by_id(rol_id):
        """Obtiene un rol por su ID"""
        sql = "SELECT * FROM RolesOrganizador WHERE id = %s"
        return Database.execute_query(sql, (rol_id,), fetch_one=True)
    
    @staticmethod
    def create(nombre):
        """Crea un nuevo rol"""
        sql = "INSERT INTO RolesOrganizador (nombre) VALUES (%s)"
        return Database.execute_query(sql, (nombre,))
    
    @staticmethod
    def update(rol_id, nombre):
        """Actualiza un rol existente"""
        sql = "UPDATE RolesOrganizador SET nombre = %s WHERE id = %s"
        return Database.execute_query(sql, (nombre, rol_id))
    
    @staticmethod
    def delete(rol_id):
        """Elimina un rol"""
        sql = "DELETE FROM RolesOrganizador WHERE id = %s"
        return Database.execute_query(sql, (rol_id,))

File: node/app/models/RolModel.js
const db = require('../database');

class RolModel {
    async findAll() {
        const [rows] = await db.query('SELECT * FROM rolesorganizador');
        return rows;
    }

    async findById(id) {
        const [rows] = await db.query('SELECT * FROM rolesorganizador WHERE id = ?', [id]);
        return rows[0];
    }

    async create(data) {
        const { nombre } = data;
        const [result] = await db.query(
            'INSERT INTO rolesorganizador (nombre) VALUES (?)',
            [nombre]
        );
        return result;
    }

    async update(id, data) {
        const { nombre } = data;
        const [result] = await db.query(
            'UPDATE rolesorganizador SET nombre = ? WHERE id = ?',
            [nombre, id]
        );
        return result;
    }

    async delete(id) {
        const [result] = await db.query('DELETE FROM rolesorganizador WHERE id = ?', [id]);
        return result;
    }
}

module.exports = new RolModel();